apiary.apib

```diff
@@ -1706,22 +1706,802 @@ root will not be listable, as it is virtual. Access known paths directly.
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/connect?arg="
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 11 Feb 2016 23:05:22 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "invalid peer address: invalid ipfs address"
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "invalid peer address: invalid ipfs address",
+          "Code": 0
+        }
+        ```
+
++ Request With Invalid Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/disconnect?arg=kitten"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/disconnect?arg=kitten"
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 11 Feb 2016 23:11:48 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "invalid peer address: invalid multiaddr, must begin with /"
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "invalid peer address: invalid multiaddr, must begin with /",
+          "Code": 0
+        }
+        ```
+
++ Request With Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/connect?arg=/ip4/40.78.22.28/tcp/4001/ipfs/QmXsaWN5ygTqHwtfoLk5gX1uvrJrYvgKtwa6TUAh4PpAKW"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/connect?arg=/ip4/40.78.22.28/tcp/4001/ipfs/QmXsaWN5ygTqHwtfoLk5gX1uvrJrYvgKtwa6TUAh4PpAKW"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 11 Feb 2016 23:07:19 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        - Strings (array)
+            - (string)
+
+    + Body
+
+        ```
+        {
+          "Strings": [
+            "connect QmXsaWN5ygTqHwtfoLk5gX1uvrJrYvgKtwa6TUAh4PpAKW success"
+          ]
+        }
+        ```
+
+## disconnect [GET /swarm/disconnect{?arg}]
+Close the connection to a given address.
```

Comment:
This just disconnects for the moment. It isn't permanent. If ipfs needs to talk to that address later, it will reconnect. Not permanent.PR this, @RichardLitt